File: src/main/java/io/github/binaryfoo/decoders/CryptogramInformationDecoder.java
package io.github.binaryfoo.decoders;

public class CryptogramInformationDecoder implements PrimitiveDecoder {

    @Override
    public String decode(String hexString) {
        CryptogramType cryptogramType = CryptogramType.fromHex(hexString);
        return cryptogramType == null ? hexString : cryptogramType.toString();
    }
}

File: src/main/java/io/github/binaryfoo/decoders/CryptogramInformationDecoder.kt
package io.github.binaryfoo.decoders

public class CryptogramInformationDecoder : PrimitiveDecoder {

    override fun decode(hexString: String): String {
        val cryptogramType = CryptogramType.fromHex(hexString)
        return if (cryptogramType == null) hexString else cryptogramType.toString()
    }
}
